### PR TITLE
FPM Toggle in Ads Settings

### PR DESCRIPTION
**Pull request by ankitrox**
## Summary

Addresses issue:

- #9655

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.
- [x] Ensure there are no unexpected significant changes to file sizes.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

**Review by benbowler — APPROVED** [review]
Tested and approved, I added a note to help QA test if healthchecks are failing.

File: assets/js/modules/ads/datastore/settings.test.js [hunk 1]
```diff
@@ -73,6 +80,38 @@ describe( 'modules/ads settings', () => {
...
+			fetchMock.postOnce( fpmEndpoint, {
+				body: JSON.stringify( {
+					data: { settings: { isEnabled: true } },
+				} ),
+				status: 200,
+			} );
+
+			await registry
+				.dispatch( CORE_SITE )
+				.setFirstPartyModeEnabled( true );
```
Comment (hussain-t) [inline assets/js/modules/ads/datastore/settings.test.js:103]: We don't need to `await` to set this local state.
```suggestion
			registry.dispatch( CORE_SITE ).setFirstPartyModeEnabled( true );
```

File: assets/js/modules/ads/datastore/settings.test.js [hunk 2]
```diff
@@ -73,6 +80,38 @@ describe( 'modules/ads settings', () => {
 				},
 			} );
 		} );
+
+		it( 'should send a POST request to FPM settings endpoint when toggle state is changed', async () => {
```
Comment (hussain-t) [inline assets/js/modules/ads/datastore/settings.test.js:84]:
```suggestion
		it( 'should send a POST request to the FPM settings endpoint when the toggle state is changed', async () => {
```

File: assets/js/modules/ads/datastore/settings.test.js [hunk 3]
```diff
@@ -73,6 +80,38 @@ describe( 'modules/ads settings', () => {
...
+			const fpmEndpoint = new RegExp(
```
Comment (hussain-t) [inline assets/js/modules/ads/datastore/settings.test.js:85]:
```suggestion
			const fpmSettingsEndpoint = new RegExp(
```

**Review by hussain-t — CHANGES_REQUESTED** [review]
Thanks, @ankitrox! Overall, the changes look good. However, I have left a few minor feedbacks. I'll address them myself to avoid a round of iteration.

**Pull request merged**